airflow/providers/amazon/aws/transfers/imap_attachment_to_s3.py

```diff
@@ -51,24 +52,26 @@ def __init__(
         self,
         *,
         imap_attachment_name: str,
+        s3_bucket: str,
         s3_key: str,
         imap_check_regex: bool = False,
         imap_mail_folder: str = 'INBOX',
         imap_mail_filter: str = 'All',
         s3_overwrite: bool = False,
         imap_conn_id: str = 'imap_default',
-        s3_conn_id: str = 'aws_default',
+        aws_conn_id: str = 'aws_default',
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
         self.imap_attachment_name = imap_attachment_name
+        self.s3_bucket = s3_bucket
         self.s3_key = s3_key
         self.imap_check_regex = imap_check_regex
         self.imap_mail_folder = imap_mail_folder
         self.imap_mail_filter = imap_mail_filter
         self.s3_overwrite = s3_overwrite
         self.imap_conn_id = imap_conn_id
-        self.s3_conn_id = s3_conn_id
+        self.aws_conn_id = aws_conn_id
```

Comment:
This is a breaking change. We can't just remove it.
We should raise deprecation warning. See for example:
https://github.com/apache/airflow/blob/602abe8394fafe7de54df7e73af56de848cdf617/airflow/providers/amazon/aws/transfers/mongo_to_s3.py#L78-L80

Reply:
Very good point! I totally miss that. Let me roll back that change. I just changed it for cosmetic details, I dont think it's necessary to raise a deprecation warning

Reply:
The change it good and needed :)
Just do with with warning

Reply:
Done